rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/Compiler.scala

```diff
@@ -12,36 +12,50 @@ import coop.rchain.rholang.interpreter.errors._
 import coop.rchain.rholang.syntax.rholang_mercury.Absyn.Proc
 import coop.rchain.rholang.syntax.rholang_mercury.{parser, Yylex}
 
-trait ParBuilder[F[_]] {
-  def buildNormalizedTerm(source: String, normalizerEnv: Map[String, Par]): F[Par]
+trait Compiler[F[_]] {
 
-  def buildNormalizedTerm(reader: Reader, normalizerEnv: Map[String, Par]): F[Par]
+  def sourceToADT(source: String): F[Par] =
+    sourceToADT(source, Map.empty[String, Par])
+
+  def sourceToADT(source: String, normalizerEnv: Map[String, Par]): F[Par] =
+    sourceToADT(new StringReader(source), normalizerEnv)
+
+  def sourceToADT(reader: Reader): F[Par] =
+    sourceToADT(reader, Map.empty[String, Par])
+
+  def sourceToADT(reader: Reader, normalizerEnv: Map[String, Par]): F[Par]
+
+  def astToADT(proc: Proc): F[Par] =
+    astToADT(proc, Map.empty[String, Par])
+
+  def astToADT(proc: Proc, normalizerEnv: Map[String, Par]): F[Par]
+
+  def sourceToAST(source: String): F[Proc] =
+    sourceToAST(new StringReader(source))
+
+  def sourceToAST(reader: Reader): F[Proc]
```

Comment:
Another good-to-have suggestion for a better modular project and future syntax sugar development though I am not sure if that's reasonable.
I think it would be better to have a method that translates syntax sugar to another AST. What I mean is that we can separate procedures into 
`sourceString -> RichAST(AST with syntax sugar) -> PureAST(AST without syntax sugar) -> ADT`.
Currently the `RichAST(AST with syntax sugar) -> PureAST(AST without syntax sugar)` methods are embedded into `astToADT`.

Reply:
That's definitely a separate PR!

Reply:
Yes.